src/indicators/money_flow_index.rs

```diff
@@ -0,0 +1,235 @@
+use std::collections::VecDeque;
+use std::fmt;
+
+use crate::errors::*;
+use crate::{Close, High, Low, Next, Reset, Volume};
+
+/// Money Flow Index (MFI).
+///
+/// The MFI is an volume and price based oscillator which gives moneyflow over n periods.
+/// MFI is used to measure buying and selling pressure.
+/// MFI is also known as volume-weighted RSI.
+///
+/// # Formula
+///
+/// Typical Price(TP) = (High + Low + Close)/3
+/// Raw Money Flow(RMF) = Typical Price x Volume
+/// RMF is positive when currnt TP is greater that previous period TP and
+/// negative when current TP is less than preivous TP
+/// Positive money flow (PMF)- calculated by adding the money flow of all the days RMF is positive
+/// Negative money flow (PMF)- calculated by adding the money flow of all the days RMF is negative
+/// Money Flow Index(MFI) = 100 * (PMF / (PMF + NMF) )
+///
+/// Where:
+///
+/// _tp_ - typical price
+/// _rmf_ - raw MF
+/// _pmf_ - positive MF
+/// _nmf_ - negative MF
+///
+/// # Parameters
+///
+/// * _n__ - number of periods, integer greater than 0
+///
+/// # Example
+///
+/// ```
+/// use ta::indicators::MoneyFlowIndex;
+/// use ta::{Next, DataItem};
+///
+/// let mut mfi = MoneyFlowIndex::new(3).unwrap();
+/// let di = DataItem::builder()
+///             .high(3.0)
+///             .low(1.0)
+///             .close(2.0)
+///             .open(1.5)
+///             .volume(1000.0)
+///             .build().unwrap();
+/// mfi.next(&di);
+///
+/// ```
+/// # Links
+/// ![Moeny Flow Index, Wikipedia](https://en.wikipedia.org/wiki/Money_flow_index)
+/// ![Money Flow Index, stockcharts] (https://stockcharts.com/school/doku.php?id=chart_school:technical_indicators:money_flow_index_mfi)
+
+#[derive(Debug, Clone)]
+pub struct MoneyFlowIndex {
+    n: u32,
+    mf_vec: VecDeque<f64>,
+    prev_tp: f64,
+}
+
+impl MoneyFlowIndex {
+    pub fn new(n: u32) -> Result<Self> {
+        match n {
+            0 => Err(Error::from_kind(ErrorKind::InvalidParameter)),
+            _ => {
+                let indicator = Self {
+                    n: n,
+                    mf_vec: VecDeque::with_capacity(n as usize + 1),
+                    prev_tp: 0.0,
+                };
+                Ok(indicator)
+            }
+        }
+    }
+}
+
+impl Next<f64> for MoneyFlowIndex {
+    type Output = f64;
+
+    fn next(&mut self, mf: f64) -> f64 {
```

Comment:
Sorry I got confused here.
From the description of the MoneyFlowIndex, I got understanding, that it is based on `volume` and price deltas. 
From this it does not look like it should be possible to implement `Next<f64>` for this indicator. Please correct me if I am wrong.

What is the meaning of `mf` here? Is it expected to be `close`, `volume` or something else? (I see that there is comparison `i < 0.0` below and it confuses me).

Reply:
I suspect that `mf` stays for `money flow`, but money flow is not something you could obtain from regular quote data. It is derivative from `close`, `low`, `high` and `volume`. So `impl Next<f64>` should be removed.
Please feel free correct me if I got it totally wrong.

Reply:
Added Next<f64>  implementation as test_helper broke. Test_helper checks for that specific implementation.
If test_helper is modified this can be removed.